### PR TITLE
[OSD-21606] update the prometheus remote write to use apiServerInternalURI from infrastructure

### PR DESCRIPTION
**Pull request by bmeng**
…nfrastructure

### What type of PR is this?
_(/feature/)_

### What this PR does / Why we need it?
With the new OCM side change, the cluster naming policy will be changed, the cluster name may not be the same as the dns name for now.
Update the prometheus remote write to match the same value as the internal api endpoint

### Which Jira/Github issue(s) does this PR fix?

_Resolves #[OSD-21606](https://issues.redhat.com//browse/OSD-21606)_

### Pre-checks (if applicable)

- [ ] Validated the changes in a hosted cluster
- [ ] Included documentation changes with PR

### CODE REVIEW

**Review by supreeth7 — CHANGES_REQUESTED** [review]
The non-uwm config needs to be modified as well.

Ref: https://github.com/openshift/hypershift-dataplane-metrics-forwarder/pull/54/files#diff-9e00c525fa5975e0d9cc35010664db0aada68d238e2c86ce7d64d8a4d24d48acR315

File: hack/templates/00-hypershift-dataplane-metrics-forwarder.SelectorSyncSet.yaml.tmpl [hunk 1]
```diff
@@ -127,7 +127,7 @@ objects:
                                   source: "DP"
                                 replicas: 2
                                 remoteWrite:
-                                  - url: https://metrics-forwarder.apps.{{ range $res := ((lookup "config.openshift.io/v1" "Infrastructure" "default" "cluster").status.platformStatus.aws.resourceTags) }}{{ if eq $res.key "api.openshift.com/name" }}{{ $res.value }}{{ end }}{{ end }}.hypershift.local
+                                  - url: {{ ( trimAll ":443" (( lookup "config.openshift.io/v1" "Infrastructure" "default" "cluster").status.apiServerInternalURI )) | replace "api" "metrics-forwarder.apps" }}
```
[the remark this answers is not in the document]
Comment (bmeng) [reply hack/templates/00-hypershift-dataplane-metrics-forwarder.SelectorSyncSet.yaml.tmpl:130]: nice catch..
yeah, I think that would be the issue, will fix the replace string

[the remark this answers is not in the document]
Comment (machi1990) [reply hack/templates/00-hypershift-dataplane-metrics-forwarder.SelectorSyncSet.yaml.tmpl]: Thanks.

**Review by supreeth7 — APPROVED** [review]
/lgtm

**Pull request merged**